src/components/SelectionList/BaseSelectionList.tsx

```diff
@@ -358,7 +360,8 @@ function BaseSelectionList<TItem extends ListItem>(
                 shouldPreventDefaultFocusOnSelectRow={shouldPreventDefaultFocusOnSelectRow}
                 // We're already handling the Enter key press in the useKeyboardShortcut hook, so we don't want the list item to submit the form
                 shouldPreventEnterKeySubmit
-                rightHandSideComponent={rightHandSideComponent}
+                // Change this because of lint
+                rightHandSideComponent={rightHandSideComponent && (typeof rightHandSideComponent === 'function' ? rightHandSideComponent({} as TItem) : rightHandSideComponent)}
```

Comment:
I think this type conflict is caused by this PR. Can you explain why?

Reply:
OK. As the fix doesn’t look straightforward, I'm not sure if we should fix it here or just let it as it is (Maybe there's an ongoing fix somewhere). I'd like to defer this to @luacmartins